lib/inc/pxp-agent/modules/apply.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef SRC_MODULES_APPLY_H_
+#define SRC_MODULES_APPLY_H_
+
+#include <pxp-agent/module.hpp>
+#include <pxp-agent/results_storage.hpp>
+#include <pxp-agent/util/bolt_module.hpp>
+#include <pxp-agent/util/purgeable.hpp>
+#include <pxp-agent/module_cache_dir.hpp>
+
+#include <leatherman/locale/locale.hpp>
+#include <leatherman/curl/client.hpp>
+
+namespace PXPAgent {
+namespace Modules {
+
+class Apply : public PXPAgent::Util::BoltModule, public PXPAgent::Util::Purgeable {
+    public:
+        Apply(const boost::filesystem::path& exec_prefix,
+              const std::vector<std::string>& master_uris,
```

Comment:
This (and the settings below) are what we need to have for the various SSL settings in the ruby-side code. Maybe the best fix is to plumb these through and use them to set Puppet's settings, rather than fighting with loading settings multiple times?

Reply:
Damn thats a really good idea. Let me see if I can do that!

Reply:
We needed to add in CRL to make this work. We have decided this is the way to go and will move forward by first requiring CRL for apply module and follow up in https://tickets.puppetlabs.com/browse/PCP-893 with honoring CRL in other modules and communications with PCP broker.